packages/web3-providers/src/cryptoscam-db/index.ts

```diff
@@ -16,17 +16,36 @@ export class CryptoScamDBAPI implements ScamWarningAPI.Provider {
         return this.bloomFilter as ScalableBloomFilter
     }
 
-    async getScamWarning(key: string): Promise<ScamWarningAPI.Info | undefined> {
+    async getScamWarning(link: string): Promise<ScamWarningAPI.Info | undefined> {
         const filter = await this.getBloomFilter()
         if (!filter) return
 
-        if (!filter.has(key)) return
+        try {
+            const url = new URL(link)
 
-        return fetchJSON<ScamWarningAPI.Info>(urlcat(baseURL, `${key}.json`))
+            if (!filter.has(url.host)) return
+
+            const result = await fetchJSON<ScamWarningAPI.Info>(urlcat(baseURL, `${url.host}.json`))
+
+            if (!result) return
+
+            if (url.pathname.toLowerCase() === new URL(result.url).pathname.toLowerCase()) {
```

Comment:
Why we need to concern about equity of these two pathnames?

Reply:
The cryptoscamdb have some data like:
```json
{
            "id": "***",
            "name": "docs.google.com",
            "url": "https://docs.google.com/forms/d/e/1FAIpQLSeRw1qALR7-rbvOKRM-Wgs1TxmAN5BC7_XzATFiHIVRuRlegQ/viewform",
}

```

We should not render plugin for `https://docs.google.com/some-other-safe-link`

Reply:
What does an empty pathname like in CryptoScamDB? Some cases may fail to detect successfully:

+ Given Link: https://scam-link.com/path-to-some-thing
+ ScamDB: `name: scam-link.com` and `url: https://scam-link.com`

We know that the given one is a scam link, but ScamDB thinks all paths under `https://scam-link.com` are unsafe.